internal/provider/meta.go

```diff
@@ -281,6 +303,47 @@ func GetClient(i interface{}, meta interface{}) (*api.Client, error) {
 	return p.GetClient(), nil
 }
 
+// IsAPISupported receives a minimum version
+// that the Vault server version should be above.
+//
+// It uses the go-version package
+// to perform a semantic version comparison, and
+// returns:
+//    - a boolean describing whether the Vault
+//      server version was above the minimum version
+//
+// This function can be used to perform semantic version comparisons
+// to conditionally enable features, or to resolve any diffs in the TF
+// state based on the Vault version.
+func IsAPISupported(meta interface{}, minVersion *version.Version) bool {
+	var p *ProviderMeta
+	switch v := meta.(type) {
+	case *ProviderMeta:
+		p = v
+	default:
+		return false
+	}
+
+	return p.IsAPISupported(minVersion)
+}
+
+func getVaultVersion(client *api.Client) (string, error) {
```

Comment:
Instead of returning the stringified version you can return a `version.Version` which implements `String()` interface

ref: https://pkg.go.dev/github.com/hashicorp/go-version#Version.String

Reply:
Updated by returning a `version.Version`